packages/analytics-browser/src/plugins/file-download-tracking.ts

```diff
@@ -0,0 +1,50 @@
+import { BrowserClient, PluginType, Event, EnrichmentPlugin } from '@amplitude/analytics-types';
+import { BrowserConfig } from '../config';
+
+const FILE_DOWNLOAD_EVENT = 'file_download';
+
+export const fileDownloadTracking = (): EnrichmentPlugin => {
+  const name = 'fileDownloadTracking';
+  const type = PluginType.ENRICHMENT;
+  const setup = async (config: BrowserConfig, amplitude?: BrowserClient) => {
+    /* istanbul ignore if */
+    if (!amplitude) {
+      // TODO: Add required minimum version of @amplitude/analytics-browser
+      config.loggerProvider.warn(
+        'File download tracking requires a later version of @amplitude/analytics-browser. File download events are not tracked.',
+      );
+      return;
+    }
+
+    const ext =
+      /\.(pdf|xlsx?|docx?|txt|rtf|csv|exe|key|pp(s|t|tx)|7z|pkg|rar|gz|zip|avi|mov|mp4|mpe?g|wmv|midi?|mp3|wav|wma)$/;
+    const links = Array.from(document.getElementsByTagName('a'));
+    links.forEach((a) => {
+      const url = new URL(a.href);
+      const result = ext.exec(url.href);
+      const fileExtension = result?.[1];
+
+      if (fileExtension) {
+        a.addEventListener('click', () => {
```

Comment:
Maybe consider adding `keypress` / `keydown` event if the download link is focused and users press `enter` key

Reply:
I tested this and it appears that `click` event is also fired hence the listener should be sufficient in this context.